File: src/oop/FileSearch/Search/Operands/AndOperand.cs
using Microsoft.Extensions.FileProviders;

namespace FileSearch.Search.Operands;

public sealed class AndOperand(IReadOnlyCollection<IMatchCriteria> predicates) : OperandCriteriaBase
{
    public override bool IsMatch(IFileInfo fileInfo, SearchOptions options)
        => predicates.All(it => it.IsMatch(fileInfo, options));
}

File: src/python/oop/FileSearch/search/operands/and_operand.py
from typing import List
from .operand_base import *
from ..predicate_base import *
from ..file_info import *
from ..search_options import *

class AndOperand(OperandBase):
    def __init__(self, criterias: List[PredicateBase]):
        self.__criterias = criterias

    def is_match(self, file: FileInfo, options: SearchOptions = None) -> bool:
        for criteria in self.__criterias:
            if not criteria.is_match(file, options):
                return False

        return True
